iceoryx_examples/automotive_soa/include/owl/kom/event_publisher_uds.inl

```diff
@@ -0,0 +1,137 @@
+
+#include "owl/kom/event_publisher_uds.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline EventPublisherUds<T>::EventPublisherUds(const core::String& service,
+                                               const core::String& instance,
+                                               const core::String& event) noexcept
+    : m_publisher({service, instance, event}, {1U, iox::NodeName_t(), true})
+    , m_instanceId(instance)
+{
+}
+
+template <typename T>
+inline std::unique_ptr<T> EventPublisherUds<T>::Allocate() noexcept
+{
+    // The proxy needs some time to discover the service and create the EventSubscriberUds with the UDS server,
+    // hence the creation of the UDS client is done here
+    if (!m_calledForTheFirstTime)
+    {
+        m_uds = std::move(iox::posix::UnixDomainSocket::create(m_instanceId, iox::posix::IpcChannelSide::CLIENT)
+                              .or_else([](auto&) {
+                                  std::cout << "Failed to create UNIX domain socket!" << std::endl;
+                                  std::terminate();
+                              })
+                              .value());
+        m_calledForTheFirstTime = true;
+    }
+
+    // Allocate the memory on the heap
+    return std::make_unique<SampleType>();
+}
+
+template <typename T>
+void EventPublisherUds<T>::Send(std::unique_ptr<SampleType> userSamplePtr) noexcept
+{
+    std::string tempBuffer;
+
+    tempBuffer.append(reinterpret_cast<char*>(&userSamplePtr->counter), 4);
+
+    auto sendTimeStampNs = userSamplePtr->sendTimestamp.time_since_epoch().count();
+
+    tempBuffer.append(reinterpret_cast<char*>(&sendTimeStampNs), 8);
+
+    constexpr uint8_t BYTES_OF_SUBPACKETS_INTEGER{4};
+    uint32_t offset = static_cast<uint32_t>(tempBuffer.size()) + BYTES_OF_SUBPACKETS_INTEGER;
+    uint32_t totalSize = userSamplePtr->payloadSizeInBytes + offset;
+    userSamplePtr->subPackets = totalSize / static_cast<uint32_t>(iox::posix::UnixDomainSocket::MAX_MESSAGE_SIZE);
+    if (totalSize % static_cast<uint32_t>(iox::posix::UnixDomainSocket::MAX_MESSAGE_SIZE) > 0)
+    {
+        userSamplePtr->subPackets += 1;
+    }
+
+    tempBuffer.append(reinterpret_cast<char*>(&userSamplePtr->subPackets), 4);
+
+    uint32_t k{0};
+    uint64_t bytesToSend{userSamplePtr->payloadSizeInBytes};
+    uint64_t messageSize{0};
+
+    // We handle the special case for the first message
+    if (bytesToSend + offset <= iox::posix::UnixDomainSocket::MAX_MESSAGE_SIZE)
+    {
+        messageSize = bytesToSend;
+    }
+    else
+    {
+        messageSize = iox::posix::UnixDomainSocket::MAX_MESSAGE_SIZE - offset;
+    }
+    for (uint32_t j = 0U; j < messageSize; j++)
+    {
+        tempBuffer.push_back(userSamplePtr->data[k++]);
+    }
+    m_uds.send(tempBuffer).or_else([](auto&) {
+        std::cerr << "Error occurred while sending on UNIX domain socket!" << std::endl;
+    });
+    tempBuffer.clear();
+    bytesToSend -= messageSize;
```

Comment:
This is not correct here. We can reduce the `bytesToSend` only when the actual `send` call was successful, otherwise we have to resend the data or abort.